### PR TITLE
RELATED: NAS-3219 nonexisting attributes in filtercontext

### PR DESCRIPTION
**Pull request by MilanLempera**
---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/common/stateInitializers.ts [hunk 1]
```diff
@@ -28,6 +30,10 @@ import { DashboardContext, PrivateDashboardContext } from "../../../types/common
 import { ObjRefMap } from "../../../../_staging/metadata/objRefMap";
 import { ExtendedDashboardWidget } from "../../../types/layoutTypes";
 import { getPrivateContext } from "../../../store/_infra/contexts";
+import { loadAvailableDisplayFormRefs } from "./loadAvailableDisplayFormRefs";
+import { PromiseFnReturnType } from "../../../types/sagas";
+import { update } from "lodash/fp";
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/common/stateInitializers.ts:35]: Please use 
```
import update from "lodash/fp/update"
```
instead, it three-shakes better. We need to imporve our ESLint/dep-cruiser setup to catch this :)

File: libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/common/stateInitializers.ts [hunk 2]
```diff
@@ -55,6 +61,43 @@ export function actionsToInitializeNewDashboard(
...
+        .filter(isDashboardAttributeFilter)
+        .map((f) => f.attributeFilter.displayForm);
+
+    const availableRefs: PromiseFnReturnType<typeof loadAvailableDisplayFormRefs> = yield call(
+        loadAvailableDisplayFormRefs,
+        ctx,
+        usedFilterDisplayForms,
+    );
+
+    return update(
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/common/stateInitializers.ts:87]: Nitpick: You could also use immer to make this more type-safe.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7008/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8874/ : FAILURE in 16m 30s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2293

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2035/ : SUCCESS in 1m 05s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2318/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1129/ : SUCCESS in 13m 31s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8880/ : SUCCESS in 16m 50s

**Pull request merged**